src/blockencodings.cpp

```diff
@@ -30,12 +30,7 @@ CBlockHeaderAndShortTxIDs::CBlockHeaderAndShortTxIDs(const CBlock& block) :
 }
 
 void CBlockHeaderAndShortTxIDs::FillShortTxIDSelector() const {
-    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
-    stream << header << nonce;
-    CSHA256 hasher;
-    hasher.Write((unsigned char*)&(*stream.begin()), stream.end() - stream.begin());
-    uint256 shorttxidhash;
-    hasher.Finalize(shorttxidhash.begin());
+    uint256 shorttxidhash = (CSHA256Writer(SER_NETWORK, PROTOCOL_VERSION) << header << nonce).GetHash();
```

Comment:
I'm gonna opt to leave this change here for now...I agree with it in principal, but would prefer to limit code churn as much as possible at this point. Can you submit it as a separate pull req later?